doc/user/client.md

```diff
@@ -0,0 +1,252 @@
+
+controller-runtime defines several interfaces used for cluster interaction:
+- `client.Client`: implementers perform CRUD operations on a k8s cluster.
+- `manager.Manager`: manages shared dependencies, such as Caches and Clients.
+- `reconcile.Reconciler`: compares provided state with actual cluster state and updates the cluster on finding state differences using a Client.
+
+## Client Usage
+
+### Default Client
+
+The SDK relies on a `manager.Manager` to create a `client.Client` interface that performs Create, Update, Delete, Get, and List operations within a `reconcile.Reconciler`'s Reconcile function. The SDK will generate code to create a Manager, which holds a Cache and a Client to be used in CRUD operations and communicate with the API server. By default a Controller's Reconciler will be populated with the Manager's Client which is a [split-client][doc-split-client].
+
+`pkg/controller/<kind>/<kind>_controller.go`:
+```Go
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileKind{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+type ReconcileKind struct {
+	// Populated above from a manager.Manager.
+	client client.Client
+	scheme *runtime.Scheme
+}
+```
+
+A split client reads (Get and List) from the Cache and writes (Create, Update, Delete) to the API server. Reading from the Cache significantly reduces request load on the API server; as long as the Cache is updated by the API server, read operations are eventually consistent. 
+
+### Non-default Client
+
+An operator developer may wish to create their own Client with different read/write optimzations, for example. controller-runtime provides a [constructor][doc-client-constr] for Clients:
+
+```Go
+// New returns a new Client using the provided config and Options.
+func New(config *rest.Config, options client.Options) (client.Client, error)
+```
+
+Creating a new Client is not usually necessary nor advised, as the default Client is sufficient for most use cases. One reason to create a custom Client is for writing tests, a situation that might require a mock Client. controller-runtime provides a [fake Client][doc-client-fake] package for this purpose.
+
+### Reconcile and the Client API
+
+A Reconciler implements the [`reconcile.Reconciler`][doc-reconcile-reconciler] interface, which exposes the Reconcile method. Reconcilers are added to a corresponding Controller for a Kind; Reconcile is called in response to cluster or external Events, with a `reconcile.Request` object argument, to read and write cluster state by the Controller, and returns a `reconcile.Result`. SDK Reconcilers have access to a Client in order to make k8s API calls.
+
+**Note**: For those familiar with the SDK's old project structure, Reconcile replaces [Handle][doc-osdk-handle].
```

Comment:
I think we might want to be a little more pedantic here. The reconciler will replace the handle function of a single object/struct/gvk. The old handler you could have `v1alpha1.MyResource` and `v1alpha1.MySecondResource` handled by the same `handle` function.

Reply:
@shawn-hurley We're going to have a separate doc highlighting the major differences between the old SDK APIs and project layout vs the controller-runtime pkgs. And how sdk.Handle() maps to reconciler.Reconcile() etc.
That's another doc that I'm working on along with the migration doc.
We can point to that doc from here.

Reply:
I agree a more detailed doc is important, I guess I was trying to say that it might be better to be really explicit here. Assuming this will be merged before the migration doc, I don't want to give the wrong impression even if we clarify later. Thoughts?

Reply:
Yeah we can be a little more explicit here and mention that Reconcile handles the reconciliation requests for a single resource type unlike the old sdk handle function which could receive events for multiple resource types.

But probably not delve more into the specific interface and why that's so(since that gets into how you watch multiple resource in the controller). Those details can be deferred to the API differences doc.

Reply:
@hasbro17 @shawn-hurley does this seem reasonable now?